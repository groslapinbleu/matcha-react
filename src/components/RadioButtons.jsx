import React, { Component } from 'react';
import PropTypes from 'prop-types'

class RadioButtons extends Component {
    constructor(props) {
        super(props)
        let index = 0
        if (props.selectedElement >= 0 && props.selectedElement < props.elementList.length) {
            index = props.selectedElement
        }
        this.state = {
            selectedElement: index
        }
        this.handleChange = this.handleChange.bind(this);
    }

    handleChange(event) {
        console.log("RadioButton handleChange")
        const value = parseInt(event.target.value, 10)
        this.setState({ selectedElement: value })
        this.props.onSelect(event.target.name, value)
    }

    render() {
        return (
            <div>
                {
                    this.props.elementList.map((element, index) => {
                        // console.log("index: "+index + " element: " + element)
                        return (
                            <div key={`RadioButton-${index}`}>
                                <label>
                                    <input
                                        className={this.props.className}
                                        type="radio" name={this.props.name}
                                        value={index}
                                        checked={this.state.selectedElement === index}
                                        onChange={this.handleChange} />
                                    {element}
                                </label>
                                <br />
                            </div>
                        )
                    })
                }
            </div>
        )
    }
}

RadioButtons.propTypes = {
    selectedElement: PropTypes.number.isRequired,
    elementList: PropTypes.array.isRequired,
    name: PropTypes.string.isRequired,
    className: PropTypes.string,
    onSelect: PropTypes.func
}
export default RadioButtons;