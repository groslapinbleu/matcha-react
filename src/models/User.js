export const defaultUserData = {
  email: '',
  username: '',
  firstname: '',
  lastname: '',
  photoURL: null,
  description: '',
  birthday: null,
  gender: 0, // see genders array below
  region: 0, // see regions array below
  preferredGender: 0,
  visible: false,
  roles: { ADMIN: false },
  friends: {}, // Format: {uid : boolean}
  tags: [], // see tags array below
};

export const genders = [
  'other', // index 0 is the default value
  'female',
  'male',
];

export const regions = [
  'other', // index 0 is the default value
  'Ile-de-France',
  'Hauts-de-France',
  'Bretagne',
  'Normandie',
  'Auvergne-Rh√¥nes-Alpes',
];

export const tags = [
  // there is no default value
  'sex',
  'love',
  'bondage',
  'romance',
  'movies',
  'literature',
  'theater',
  'cooking',
  'kids',
  'money',
];

export function isBFriendOfA(userA, userB) {
  if (userA.friends && userB.uid in userA.friends) {
    if (userA.friends[userB.uid]) {
      return true;
    } else return false;
  }
  return null;
}
