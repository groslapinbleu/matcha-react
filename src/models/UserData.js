export const defaultUserData = {
        email : "",
        username : "",
        firstname: "",
        lastname: "",
        photoURL : null, 
        description : "",
        gender : 0,
        region : 0,
        preferredGender : 0,
        visible : false,
        roles : {ADMIN: false}
    }

    export const genders = [
        "other",
        "female",
        "male"
    ]

    export const regions = [
        "other",
        "Ile-de-France",
        "Hauts-de-France",
        "Bretagne",
        "Normandie",
        "Auvergne-Rh√¥nes-Alpes"
    ]