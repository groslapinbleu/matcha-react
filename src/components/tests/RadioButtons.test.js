import RadioButtons from 'components/RadioButtons';
import React from 'react';
import renderer from 'react-test-renderer';
import { genders } from 'models/User';

const handleSelect = (name, value) => {
  console.log('name =' + name + ' value = ' + value);
};

test('RadioButtons snapshot test', () => {
  const component = renderer.create(
    <RadioButtons
      selectedElement={0}
      elementList={genders}
      name='test'
      className='border-indigo-300'
      onSelect={handleSelect}
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
