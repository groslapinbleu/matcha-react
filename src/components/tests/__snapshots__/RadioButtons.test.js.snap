// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtons snapshot test 1`] = `
<div>
  <div>
    <label>
      <input
        checked={true}
        className="border-indigo-300"
        name="test"
        onChange={[Function]}
        type="radio"
        value={0}
      />
      autre
    </label>
    <br />
  </div>
  <div>
    <label>
      <input
        checked={false}
        className="border-indigo-300"
        name="test"
        onChange={[Function]}
        type="radio"
        value={1}
      />
      femme
    </label>
    <br />
  </div>
  <div>
    <label>
      <input
        checked={false}
        className="border-indigo-300"
        name="test"
        onChange={[Function]}
        type="radio"
        value={2}
      />
      homme
    </label>
    <br />
  </div>
</div>
`;
